address_book.rb

```diff
@@ -0,0 +1,293 @@
+require "yaml"
+require 'csv'
+require_relative 'person'
+require_relative 'address'
+class AddressBook
+
+   @@contacts=Array.new
+
+   def open
+     if File.exist?("contacts.yml")
+       @@contacts = YAML.load_file("contacts.yml")
+     end
+
+=begin
+     if File.exist?("contacts.csv")
+     @@contacts = CSV.read('contactBook.csv')
+   end #def (open)
+=end
+end
+   def save
+     File.open("contacts.yml", "w") do |file|
+       file.write(@@contacts.to_yaml)
+     end #file.open
+=begin
+     File.open("contactBook.csv", "w") do |file|
+       for i in 0..@@contacts.length-1
+       file.write(@@contacts[i])
+       file.write(@@contacts[i].getfirstName)
+       end
+     end #file.open
+=end
+   end
+
+ def isAlpha(word)
+   if /\d/.match( word)
+     return false
+   else
+     return true
+   end
+end
+
+
+   def isNum(num)
+     if /\d/.match( num)
+        return true
+     else
+      return false
+
+     end
+   end
```

Comment:
Since the above methods are generic check for the regular expression, can you please try to write the above methods in a common module?